### ISSUE TITLE
Fragmented sentences in translatable strings

### ISSUE THREAD

**Issue by aitap** [Rdatatable/data.table#6482]
Splitting a sentence into fragments may make it harder to translate because the parts and the whole have different grammatical cases. 

For example, [here](https://github.com/Rdatatable/data.table/blob/cf05b4a0951958d2b6f9f7a4819e5b2cd851ceca/src/assign.c#L739) "target vector" would be "целево**й** вектор", but [later](https://github.com/Rdatatable/data.table/blob/cf05b4a0951958d2b6f9f7a4819e5b2cd851ceca/src/assign.c#L787) "Assigning factor numbers to `%s`", "target vector" must be "Присваиваю фактор целево**му** вектор**у**". I think that this is limited to `targetDesc()` in `assign.c`, all such cases are in the dative case and so can be translated into Russian pre-inflected, but I'm not sure this would remain the case for Arabic and Hindi.

Also a problem for languages with strict word order might be concatenation of messages in `fread.c` [[1](https://github.com/Rdatatable/data.table/blob/cf05b4a0951958d2b6f9f7a4819e5b2cd851ceca/src/fread.c#L446), [2](https://github.com/Rdatatable/data.table/blob/cf05b4a0951958d2b6f9f7a4819e5b2cd851ceca/src/fread.c#L451)], used [here](https://github.com/Rdatatable/data.table/blob/cf05b4a0951958d2b6f9f7a4819e5b2cd851ceca/src/fread.c#L1537). [This](https://github.com/Rdatatable/data.table/blob/cf05b4a0951958d2b6f9f7a4819e5b2cd851ceca/src/fread.c#L1804), on the other hand, is already slightly broken because `msg` is a full sentence.

_Originally posted by @aitap in https://github.com/Rdatatable/data.table/issues/6194#issuecomment-2333894555_

**Comment by rikivillalba** [comment]
Can those be enumerated in some site i.e. a branch in which those are flagged with a comment // [fragmented] or alike?

**Comment by MichaelChirico** [comment]
The `targetDesc()` case in assign.c is tough. Basically, there are a series of operations that can either affect a simple vector, or a "special" vector, namely one where we have extra info that it's a certain column in a data.frame. When we have that extra info, it's nice to present to the user, otherwise generic "target vector" is the description. It's used for 7 messages now:

https://github.com/Rdatatable/data.table/blob/c616cb9c9a37ad7d3adfcd1e3fedda2c3b1e527a/src/assign.c#L789
https://github.com/Rdatatable/data.table/blob/c616cb9c9a37ad7d3adfcd1e3fedda2c3b1e527a/src/assign.c#L798
https://github.com/Rdatatable/data.table/blob/c616cb9c9a37ad7d3adfcd1e3fedda2c3b1e527a/src/assign.c#L890
https://github.com/Rdatatable/data.table/blob/c616cb9c9a37ad7d3adfcd1e3fedda2c3b1e527a/src/assign.c#L897
https://github.com/Rdatatable/data.table/blob/c616cb9c9a37ad7d3adfcd1e3fedda2c3b1e527a/src/assign.c#L902
https://github.com/Rdatatable/data.table/blob/c616cb9c9a37ad7d3adfcd1e3fedda2c3b1e527a/src/assign.c#L910
https://github.com/Rdatatable/data.table/blob/c616cb9c9a37ad7d3adfcd1e3fedda2c3b1e527a/src/assign.c#L924

I don't really see a simple way forward that balances code readability with general translatability -- any suggestions?

@aitap, do I understand that the key is the grammatical case? Maybe we can use a 3rd argument `int case` in `targetDesc()` to distinguish "to {target vector,column %d}" from "of {target vector, column %d}", is that sufficient?

**Comment by aitap** [comment]
@rikivillalba So far I've only found `targetDesc()` and `copyFile()` to pose problems like this. We should be able to find the rest of the cases (if they exist) as we go through the translatable strings and their context.

@MichaelChirico Very good idea to combine with a preposition, thank you!

I've experimented a bit with automatic translation to Arabic and Hindi and I can't see any differences in inflection of the rest of the sentence (`Assigning factor numbers to %s`) depending on whether it's a vector or the column. So solving the grammatical case problem could be just enough.
